examples/preprocessing/plot_all_scaling.py

```diff
@@ -0,0 +1,328 @@
+y = minmax_scale(y_full)
+
+
+def create_axes(title, figsize=(16, 6)):
+    fig = plt.figure(figsize=figsize)
+    fig.suptitle(title)
+
+    # define the axis for the first plot
+    left, width = 0.1, 0.22
+    bottom, height = 0.1, 0.7
+    bottom_h = height + 0.15
+    left_h = left + width + 0.02
+
+    rect_scatter = [left, bottom, width, height]
+    rect_histx = [left, bottom_h, width, 0.1]
+    rect_histy = [left_h, bottom, 0.05, height]
+
+    ax_scatter = plt.axes(rect_scatter)
+    ax_histx = plt.axes(rect_histx)
+    ax_histy = plt.axes(rect_histy)
+
+    # define the axis for the zoomed-in plot
+    left = width + left + 0.2
+    left_h = left + width + 0.02
+
+    rect_scatter = [left, bottom, width, height]
+    rect_histx = [left, bottom_h, width, 0.1]
+    rect_histy = [left_h, bottom, 0.05, height]
+
+    ax_scatter_zoom = plt.axes(rect_scatter)
+    ax_histx_zoom = plt.axes(rect_histx)
+    ax_histy_zoom = plt.axes(rect_histy)
+
+    # define the axis for the colorbar
+    left, width = width + left + 0.13, 0.01
+
+    rect_colorbar = [left, bottom, width, height]
+    ax_colorbar = plt.axes(rect_colorbar)
+
+    return ((ax_scatter, ax_histy, ax_histx),
+            (ax_scatter_zoom, ax_histy_zoom, ax_histx_zoom),
+            ax_colorbar)
+
+
+def plot_distribution(axes, X, y, hist_nbins=50, title="",
+                      x0_label="", x1_label=""):
+    ax, hist_X1, hist_X0 = axes
+
+    ax.set_title(title)
+    ax.set_xlabel(x0_label)
+    ax.set_ylabel(x1_label)
+
+    # The scatter plot
+    colors = cm.plasma_r(y)
+    ax.scatter(X[:, 0], X[:, 1], alpha=0.5, marker='o', s=5, lw=0, c=colors)
+
+    # Removing the top and the right spine for aesthetics
+    # make nice axis layout
+    ax.spines['top'].set_visible(False)
+    ax.spines['right'].set_visible(False)
+    ax.get_xaxis().tick_bottom()
+    ax.get_yaxis().tick_left()
+    ax.spines['left'].set_position(('outward', 10))
+    ax.spines['bottom'].set_position(('outward', 10))
+
+    # Histogram for axis X1 (feature 5)
+    hist_X1.set_ylim(ax.get_ylim())
+    hist_X1.hist(X[:, 1], bins=hist_nbins, orientation='horizontal',
+                 color='grey', ec='grey')
+    hist_X1.axis('off')
+
+    # Histogram for axis X0 (feature 0)
+    hist_X0.set_xlim(ax.get_xlim())
+    hist_X0.hist(X[:, 0], bins=hist_nbins, orientation='vertical',
+                 color='grey', ec='grey')
+    hist_X0.axis('off')
+
+###############################################################################
+# Two plots will be shown for each scaler/normalizer/transformer. The left
+# figure will show a scatter plot of the full data set while the right figure
+# will exclude the extreme values considering only 99 % of the data set,
+# excluding marginal outliers. In addition, the marginal distributions for each
+# feature will be shown on the side of the scatter plot.
+
+
+def make_plot(item_idx):
+    title, X = distributions[item_idx]
```

Comment:
can we get titles for the two subplots? One is the full data, the other a zoom-in, right? That's not clear from the figure.